src/app/helpers/formTemplate/deleteTemplate.ts

```diff
@@ -0,0 +1,18 @@
+import { Observable } from "rxjs";
+import { AbstractModel } from "src/app/models/AbstractModel";
+import { FormTemplate } from "./formTemplate";
+
+export abstract class DeleteFormTemplate<
+  M extends AbstractModel
+> extends FormTemplate<M, {}> {
+  /**
+   * Create delete request
+   * @param event Form submission
+   */
+  protected abstract apiDestroy(): Observable<M | void>;
```

Comment:
Why do we need this?

The API has a standard destroy method, as in the interface.

Is this a before/after hook?